### ISSUE TITLE
bibstyle and bib in a newcommand won't run bibtex automatically 

### ISSUE THREAD

**Issue by syvshc** [texstudio-org/texstudio#1764]
### Environment
- TeXstudio: 3.1.2
- Qt: 5.12.2
- OS: ArchLinux 5.14.7-arch1-1, Windows10
- TeX distribution: texlive2021

### Expected behavior

Here is my mwe:
```latex
\documentclass{article}
\newcommand{\Bib}[2]{
    \bibliographystyle{#1}
    \bibliography{#2}
}
\begin{document}
    \cite{article-full}
    \Bib{plain}{xampl.bib}
\end{document}
```
Here I put `\bibliographystyle` and `\bibliography` in a `\newcommand`, I press F5 to start compile, but it didn't run `bibtex` automatically, I have to run `bibtex` manually and then F5

![texstudio](https://user-images.githubusercontent.com/38098591/134685234-30e73700-1d51-4d70-9e4b-981dce0d0db3.gif)

If I write like this:
```latex
\documentclass{article}
%\newcommand{\Bib}[2]{
%    \bibliographystyle{#1}
%    \bibliography{#2}
%}
\bibliographystyle{plain}
\begin{document}
    \cite{article-full}
%    \Bib{plain}{xampl.bib}
	\bibliography{xampl.bib}
\end{document}
```
F5 can do everything well

![texstudioright](https://user-images.githubusercontent.com/38098591/134685485-dd2e3ba3-d702-4ee7-bcda-3a2967b428ad.gif)

**Comment by sunderme** [comment]
any particular reason to make life hard for the parser ?

**Comment by sunderme** [comment]
on a general not. Using latexmk instead of txs functionality (with pdflatex and several commands) should work better as the tool is specialized and optimized for this .

**Comment by syvshc** [comment]
> on a general not. Using latexmk instead of txs functionality (with pdflatex and several commands) should work better as the tool is specialized and optimized for this .

Well, I know `latexmk` is better. However this is a loss of this function isn't it?

**Comment by sunderme** [comment]
more or less. Apparently txs scans for bibliography entries to trigger bibtex. You can make the user-command known to txs via a cwl file. (see wiki)
txs does not interpret user defined commands (and probably never will).
as there are several ways to cope, I don't see to introduce a special code path for this.

**Comment by syvshc** [comment]
> more or less. Apparently txs scans for bibliography entries to trigger bibtex. You can make the user-command known to txs via a cwl file. (see wiki)
> txs does not interpret user defined commands (and probably never will).
> As there are several ways to cope, I don't see the need to introduce a special code path for this.

Well, I thought that txs's multiple compiling and checking bib/ref are handle like `latexmk`, I mean that txs can watch file changing and make the right process. But it seems not?

**Comment by sunderme** [comment]
I would have to dig to see how exactly it works. But it is a bit futile to recreate an existing program like latexmk.

**Comment by syvshc** [comment]
you're right, it's different from latexmk. I'll close this issue and thanks for your reply :+1:

**Comment by muzimuzhi** [comment]
> I would have to dig to see how exactly it works.

@sunderme Partial findings: `latexmk` detects whether to run bibtex/biber by parsing `.aux` file for `\bibdata`, `\bibstyle`, and `\@input`. See [related source lines](http://www.tug.org/svn/texlive/trunk/Master/texmf-dist/scripts/latexmk/latexmk.pl?view=markup&pathrev=59535#l5677).
